src/lib/viewers/archive/ArchiveExplorer.js

```diff
@@ -190,7 +190,19 @@ class ArchiveExplorer extends React.Component {
             return 0;
         });
 
-        return sortDirection === SortDirection.DESC ? sortedItems : sortedItems.reverse();
+        if (sortDirection === SortDirection.DESC) {
+            sortedItems.reverse();
+        }
+
+        // folders always come before files
+        sortedItems.sort((a, b) => {
```

Comment:
It seems like we should be able to collapse all of this into a single `.sort` call, with `sortDirection` controlling the comparator logic. Can we also check to see how webapp does this sorting? I don't see an equivalent complex sort function there.